src/params/str-list.js

```diff
@@ -1,14 +1,26 @@
+const assert = require('assert');
 const List = require('./list');
 
 class StrList extends List {
-  constructor(...args) {
-    super(...args);
+  constructor(name, position, opts) {
```

Comment:
This removes that `opts !== undefined` below
```suggestion
  constructor(name, position, opts = {}) {
```

Reply:
I like this change